src/components/Delegate/StakingOpportunities.tsx

```diff
@@ -86,25 +89,41 @@ export const StakingOpportunities = ({ assetId, accountId }: StakingOpportunitie
 
   const marketData = useAppSelector(state => selectMarketDataById(state, assetId))
   const history = useHistory()
-  const filter = useMemo(() => ({ assetId, accountId }), [assetId, accountId])
 
-  // this is returning data grouped by validator, not by account
-  const stakingOpportunitiesData = useAppSelector(s =>
-    selectStakingOpportunitiesDataFullByFilter(s, filter),
+  const userStakingOpportunitiesFilter = useMemo(
+    () => ({
+      accountId: accountId ?? '',
+    }),
+    [accountId],
+  )
+  const userStakingOpportunities = useAppSelector(state =>
+    selectUserStakingOpportunitiesWithMetadataByFilter(state, userStakingOpportunitiesFilter),
   )
-  const hasActiveStaking = useAppSelector(state => selectHasActiveStakingOpportunity(state, filter))
+  const hasActiveStaking = userStakingOpportunities.some(userStakingOpportunity => {
+    if (!userStakingOpportunity) return false
+    const { stakedAmountCryptoBaseUnit, rewardsAmountsCryptoBaseUnit } = userStakingOpportunity
+    const undelegations = makeTotalUndelegations(userStakingOpportunity)
```

Comment:
since we know we're in a delegations component, we can guard and return false then just pass in undelegations